### PR TITLE
thunderbird-128: init at 128.0esr (+refactor)

### PR DESCRIPTION
**Pull request by vcunat**
https://blog.thunderbird.net/2024/07/welcome-to-thunderbird-128-nebula/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] N/A? Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] N/A Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by vcunat** [comment]
This doesn't switch defaults, so I assume we can merge fast.  That way people will be able to try with binaries from the official cache.

**Comment by vcunat** [comment]
I glanced into the output of configure phase and noticed nothing suspicious (like missing new stuff).

File: pkgs/applications/networking/mailreaders/thunderbird/packages.nix [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
     application = "comm/mail";
     applicationName = "Mozilla Thunderbird";
     binaryName = pname;
     src = fetchurl {
       url = "mirror://mozilla/thunderbird/releases/${version}/source/thunderbird-${version}.source.tar.xz";
-      sha512 = "182f35e8e5ece98d18dfefe106c73bc97fbc619f59772d9b3455b7c8af412021ecc5eae97a12515224e91deb814abb7a6ef7f538c450e9e77fdfd84078678038";
+      inherit sha512;
     };
     extraPatches = [
       # The file to be patched is different from firefox's `no-buildconfig-ffx90.patch`.
```
Comment (philiptaron) [inline pkgs/applications/networking/mailreaders/thunderbird/packages.nix:15]: Does this still need to be patched? Filename references 115.

File: pkgs/applications/networking/mailreaders/thunderbird/packages.nix [hunk 2]
```diff
@@ -51,6 +45,30 @@ rec {
       })];
     });
   };
```
Comment (philiptaron) [inline pkgs/applications/networking/mailreaders/thunderbird/packages.nix:47]: https://bugzilla.mozilla.org/show_bug.cgi?id=1790071 is closed two years ago; this doesn't seem relevant for 128?

File: pkgs/applications/networking/mailreaders/thunderbird/update.nix [hunk 3]
```diff
@@ -3,5 +3,5 @@
 }@args:
 
 callPackage ../../browsers/firefox/update.nix ({
-  baseUrl = "http://archive.mozilla.org/pub/thunderbird/releases/";
+  baseUrl = "https://archive.mozilla.org/pub/thunderbird/releases/";
 } // (builtins.removeAttrs args ["callPackage"]))
```
Comment (philiptaron) [inline pkgs/applications/networking/mailreaders/thunderbird/update.nix:7]: Unrelated to this patch: why this line?!

File: pkgs/applications/networking/mailreaders/thunderbird/packages.nix (hunk 2 above)
Comment (vcunat) [reply pkgs/applications/networking/mailreaders/thunderbird/packages.nix:47]: By my reading they just patched their vendored copy of ICU.  We could choose to use the vendored one.  The upstream ticket seems unresolved: https://unicode-org.atlassian.net/browse/ICU-22132

File: pkgs/applications/networking/mailreaders/thunderbird/packages.nix (hunk 1 above)
Comment (vcunat) [reply pkgs/applications/networking/mailreaders/thunderbird/packages.nix:15]: I haven't checked (yet), but it feels natural that "build config" would contain various references that we don't want (saved build flags or something).

**Review by philiptaron — APPROVED** [review]
Sounds good.

File: pkgs/applications/networking/mailreaders/thunderbird/packages.nix (hunk 1 above)
Comment (vcunat) [reply pkgs/applications/networking/mailreaders/thunderbird/packages.nix:15]: So I tried.  The difference in closure size for the unwrapped thunderbird is 0.72G vs. 4.6G.  Let me drop the confusing number from the filename.

Comment (vcunat) [reply pkgs/applications/networking/mailreaders/thunderbird/packages.nix:15]: It's rust, gcc, two clangs and more ... insane bloat.

**Pull request merged**